### PR TITLE
Method retainAll() with extensive comments.

### PR DESCRIPTION
**Pull request by udodikdmitro**

### CODE REVIEW

File: src/package1/CustomArrayList.java [hunk 1]
```diff
@@ -124,31 +124,58 @@ public boolean addAll(int index, Collection<? extends String> c) {
...
+            if (c.contains(elements[i])) {
+                arrayIndex = i;
+                break;
+            }
+        //Examination of ending and exit of method.
+            // It means that collection hasn't coincidences.
+            // Create an empty array.
+            if (i == size - 1) {
+                elements = new String[size];
+                size = 0;
```
Comment (ezseeton) [inline src/package1/CustomArrayList.java]: please use method clear instead

File: src/package1/CustomArrayList.java [hunk 2]
```diff
@@ -124,31 +124,58 @@ public boolean addAll(int index, Collection<? extends String> c) {
...
+//        4. В іншому циклі перевіряти на збіги і переприсвоювати елемент,
+//        провіряючи його на предмет збігу.
+        for (int i = 0; i < size; i++) {
+            // Examination for containing of elements
+            // and getting a start index of appropriation.
+            if (c.contains(elements[i])) {
+                arrayIndex = i;
+                break;
+            }
+        //Examination of ending and exit of method.
```
Comment (ezseeton) [inline src/package1/CustomArrayList.java]: use "checking of" instead of "Examination for" everywhere

**Review by ezseeton — COMMENTED** [review]
Your PR is awesome

File: src/package1/CustomArrayList.java [hunk 3]
```diff
@@ -124,31 +124,52 @@ public boolean addAll(int index, Collection<? extends String> c) {
...
+        elements[0] = elements[arrayIndex];
+        int newArrayIndex = 1;
+
+
+        // We're going into a remainder of previous thisArray by using a counter "i".
+        // We assign the elements are contained in a collection for a new thisArray.
+        for (int i = arrayIndex + 1; i < size; i++) {
+            if (c.contains(elements[i])) {
+                elements[newArrayIndex] = elements[i];
+                newArrayIndex++;
```
Comment (udodikdmitro) [inline src/package1/CustomArrayList.java:166]: newArrayIndex is equals or less of "i" and elements of array copy into equals or less position always.

File: src/package1/CustomArrayList.java [hunk 4]
```diff
@@ -124,31 +124,52 @@ public boolean addAll(int index, Collection<? extends String> c) {
...
+            if (i == size - 1) {
+                clear();
+                return true;
             }
         }
-        elements = a;
+        // We assign a first element to thisArray if the method comes here.
+        //Create a new counter of a new thisArray.
+        elements[0] = elements[arrayIndex];
+        int newArrayIndex = 1;
```
Comment (udodikdmitro) [inline src/package1/CustomArrayList.java:158]: newArrayIndex is equals or less of "i" and elements of array copy into equals or less position of array always.

File: src/package1/CustomArrayList.java [hunk 5]
```diff
@@ -124,31 +124,52 @@ public boolean addAll(int index, Collection<? extends String> c) {
...
+        for (int i = arrayIndex + 1; i < size; i++) {
+            if (c.contains(elements[i])) {
+                elements[newArrayIndex] = elements[i];
+                newArrayIndex++;
+            }
+        }
+        // We are copy suitable elements to new thisArray
+        // and define the number of elements for output.
+        size = newArrayIndex;
+        System.arraycopy(elements, 0, elements, 0, size);
```
Comment (udodikdmitro) [inline src/package1/CustomArrayList.java:172]: A size of thisArray is equals of value "size". Array has not empty positions.

**Review by udodikdmitro — COMMENTED** [review]
Is this a good idea: array has not empty positions?

**Comment by udodikdmitro** [comment]
Show me my mistakes in English please.

**Review by ezseeton — APPROVED** [review]

File: src/package1/CustomArrayList.java [hunk 6]
```diff
@@ -124,31 +124,52 @@ public boolean addAll(int index, Collection<? extends String> c) {
...
+            // and getting a start index of appropriation.
+            if (c.contains(elements[i])) {
+                arrayIndex = i;
+                break;
+            }
+            // Checking of ending and exit of method.
+            // It means that collection hasn't coincidences.
+            // Create an empty array.
+            if (i == size - 1) {
+                clear();
```
Comment (udodikdmitro) [inline src/package1/CustomArrayList.java:151]: I

**Pull request merged**